sandbox/host/src/sandbox/wasmer_backend.rs

```diff
@@ -309,43 +325,64 @@ pub fn instantiate(
                     .ok_or(InstantiationError::ModuleDecoding)?;
 
                 let function = match version {
-                    Instantiate::Version1 => {
-                        dispatch_function(supervisor_func_index, &context.store, func_ty)
-                    }
-                    Instantiate::Version2 => {
-                        dispatch_function_v2(supervisor_func_index, &context.store, func_ty)
-                    }
+                    Instantiate::Version1 => dispatch_function(
+                        supervisor_func_index,
+                        &mut context.store().borrow_mut(),
+                        func_ty,
+                    ),
+                    Instantiate::Version2 => dispatch_function_v2(
+                        supervisor_func_index,
+                        &mut context.store().borrow_mut(),
+                        func_ty,
+                    ),
                 };
 
-                let exports = exports_map.entry(import.module().to_string()).or_default();
-
                 exports.insert(import.name(), sandbox_wasmer::Extern::Function(function));
             }
         }
     }
 
-    let mut import_object = sandbox_wasmer::ImportObject::new();
-    for (module_name, exports) in exports_map.into_iter() {
-        import_object.register(module_name, exports);
-    }
+    let mut import_object = sandbox_wasmer::Imports::new();
+    import_object.register_namespace("env", exports);
+
+    // This is sound since we're only instantiating the module and `start` function is restricted by our code checks.
```

Comment:
Do you mean that there's nothing we actually run so we don't need any context because nothing will access this global value etc?

Reply:
Yes, that's correct. No code can access globals or call syscalls, as the start function is not allowed by Code checks. Therefore, this context loads unnecessarily.

It seems both of these can be removed at module instantiation:
```
let instance = SandboxContextStore::using(sandbox_context, || {
        dispatch_function_env::using(&mut dispatch_func_env, || {
```

Reply:
Could you extend comment a bit to avoid future misunderstanding please?

Reply:
Extended comment